spark-knn-core/src/main/scala/org/apache/spark/ml/knn/KNN.scala

```diff
@@ -228,12 +265,16 @@ class KNNModel private[ml](
   def setBufferSize(value: Double): this.type = set(bufferSize, value)
 
   //TODO: All these can benefit from DataSet API in Spark 1.6
-  override def transform(dataset: DataFrame): DataFrame = {
-    val merged = transform(dataset, topTree, subTrees)
+  override def transform(dataset: Dataset[_]): DataFrame = transform(dataset, withDistance = false)
```

Comment:
We can definitely go with a `withDistance: Option[boolean]`
I just <went with the Dummy in order to keep all previous code working, but if we use an Option, we'll achieve the same thing